### ISSUE TITLE
Pillow considers image is truncated while image displays perfectly anywhere else

### ISSUE THREAD

**Issue by JivanRoquet** [python-pillow/Pillow#4414]
Using Python 3.7 and Pillow 7.0.0

Loading a collection of images with the parameter `ImageFile.LOAD_TRUNCATED_IMAGES = False` causes the error `OSError: broken data stream when reading image file`.

When `LOAD_TRUNCATED_IMAGES` is set to True, loading is permitted but plotting the images shows only plain black rectangles.

Basically, Pillows considers that these images are truncated, while the same images are displayed perfectly well in MacOS Finder, or in Google Chrome, or in Preview, or in Photoshop, or in any other image-displaying software.

An example of the images in question:

![vneck-oaX2uoJail](https://user-images.githubusercontent.com/6452675/74116678-8ec5b080-4bac-11ea-935f-260496db3b04.jpg)

As you can observe, this image displays perfectly fine here, and it was copy/pasted directly from the source which triggers an error when loaded from Pillow.

Any information on how to solve this issue would be greatly appreciated.

**Comment by JivanRoquet** [comment]
Traceback of the error, which happens when loading images using Keras (which itself uses Pillow):

```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-119-4ea6f06895d5> in <module>
----> 1 imgs, labels = next(training_batches)
      2 plots(imgs[:10], titles=labels[:10])

/usr/local/lib/python3.7/site-packages/keras_preprocessing/image/iterator.py in __next__(self, *args, **kwargs)
    102 
    103     def __next__(self, *args, **kwargs):
--> 104         return self.next(*args, **kwargs)
    105 
    106     def next(self):

/usr/local/lib/python3.7/site-packages/keras_preprocessing/image/iterator.py in next(self)
    114         # The transformation of images is not under thread lock
    115         # so it can be done in parallel
--> 116         return self._get_batches_of_transformed_samples(index_array)
    117 
    118     def _get_batches_of_transformed_samples(self, index_array):

/usr/local/lib/python3.7/site-packages/keras_preprocessing/image/iterator.py in _get_batches_of_transformed_samples(self, index_array)
    228                            color_mode=self.color_mode,
    229                            target_size=self.target_size,
--> 230                            interpolation=self.interpolation)
    231             x = img_to_array(img, data_format=self.data_format)
    232             # Pillow images should be closed after `load_img`,

/usr/local/lib/python3.7/site-packages/keras_preprocessing/image/utils.py in load_img(path, grayscale, color_mode, target_size, interpolation)
    130                         ", ".join(_PIL_INTERPOLATION_METHODS.keys())))
    131             resample = _PIL_INTERPOLATION_METHODS[interpolation]
--> 132             img = img.resize(width_height_tuple, resample)
    133     return img
    134 

/usr/local/lib/python3.7/site-packages/PIL/Image.py in resize(self, size, resample, box, reducing_gap)
   1856             return im.convert(self.mode)
   1857 
-> 1858         self.load()
   1859 
   1860         if reducing_gap is not None and resample != NEAREST:

/usr/local/lib/python3.7/site-packages/PIL/ImageFile.py in load(self)
    268         if not self.map and not LOAD_TRUNCATED_IMAGES and err_code < 0:
    269             # still raised if decoder fails to return anything
--> 270             raise_ioerror(err_code)
    271 
    272         return Image.Image.load(self)

/usr/local/lib/python3.7/site-packages/PIL/ImageFile.py in raise_ioerror(error)
     57     if not message:
     58         message = "decoder error %d" % error
---> 59     raise OSError(message + " when reading image file")
     60 
     61 

OSError: broken data stream when reading image file
```

**Comment by radarhere** [comment]
Strangely enough, if I run the following code over your image on my macOS machine, I don't see any problems in either case. Could you run this code and report back if it works for you?

```python
from PIL import Image
im = Image.open('test.jpg')
im.load()
im.save('out.jpg')

from PIL import ImageFile
ImageFile.LOAD_TRUNCATED_IMAGES = True
im = Image.open('test.jpg')
im.load()
im.save('out2.jpg')
```

**Comment by JivanRoquet** [comment]
Thanks for the comment. Just typed this and the bug happened:

```python
im = Image.open('{}/vneck/vneck-Oagos9qua3.jpg'.format(neck_images_root))
im.load()
```

Traceback:
```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-126-acab85a2cd71> in <module>
      1 im = Image.open('{}/vneck/vneck-Oagos9qua3.jpg'.format(neck_images_root))
----> 2 im.load()

/usr/local/lib/python3.7/site-packages/PIL/ImageFile.py in load(self)
    268         if not self.map and not LOAD_TRUNCATED_IMAGES and err_code < 0:
    269             # still raised if decoder fails to return anything
--> 270             raise_ioerror(err_code)
    271 
    272         return Image.Image.load(self)

/usr/local/lib/python3.7/site-packages/PIL/ImageFile.py in raise_ioerror(error)
     57     if not message:
     58         message = "decoder error %d" % error
---> 59     raise OSError(message + " when reading image file")
     60 
     61 

OSError: broken data stream when reading image file
```

**Comment by JivanRoquet** [comment]
If I set `LOAD_TRUNCATED_IMAGES` to True:

```python
ImageFile.LOAD_TRUNCATED_IMAGES = True
im = Image.open('{}/vneck/vneck-ofeerae6Ea.jpg'.format(neck_images_root))
im.load()
im.save('out.jpg')
```

Traceback:
```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-131-0c5163ee485f> in <module>
      2 im = Image.open('{}/vneck/vneck-ofeerae6Ea.jpg'.format(neck_images_root))
      3 im.load()
----> 4 im.save('out.jpg')

/usr/local/lib/python3.7/site-packages/PIL/Image.py in save(self, fp, format, **params)
   2100 
   2101         try:
-> 2102             save_handler(self, fp, filename)
   2103         finally:
   2104             # do what we can to clean up

/usr/local/lib/python3.7/site-packages/PIL/JpegImagePlugin.py in _save(im, fp, filename)
    764     bufsize = max(ImageFile.MAXBLOCK, bufsize, len(exif) + 5, len(extra) + 1)
    765 
--> 766     ImageFile._save(im, fp, [("jpeg", (0, 0) + im.size, 0, rawmode)], bufsize)
    767 
    768 

/usr/local/lib/python3.7/site-packages/PIL/ImageFile.py in _save(im, fp, tile, bufsize)
    528                 s = e.encode_to_file(fh, bufsize)
    529             if s < 0:
--> 530                 raise OSError("encoder error %d when writing image file" % s)
    531             e.cleanup()
    532     if hasattr(fp, "flush"):

OSError: encoder error -2 when writing image file
```

**Comment by JivanRoquet** [comment]
Update: it looks like there is something corrupt in Pillow installation because the above behaviour can be reproduced for every single image, wherever it comes from.

Basically Pillow can't load images at all at the moment, on this installation.

**Comment by radarhere** [comment]
According to my e-mail, you made another comment after those two, but it's not showing on this thread. I've quoted it here. If you deleted it because it isn't correct, just let us know.

> Update: it looks like there is something corrupt in Pillow installation because the above behaviour can be reproduced for every single image, wherever it comes from.
>
> Basically Pillow can't load images at all at the moment, on this installation.

**Comment by JivanRoquet** [comment]
Yep, I deleted it because it's not correct, my mistake.

The truth is that Pillow will open any png images, but not jpg images, from what I've tried. And if I convert the problematic jpg images into png (via any application), then Pillow will open these png just fine.

**Comment by radarhere** [comment]
Can you tell us what versions of libjpeg or OpenJPEG you're running?

```python
>>> from PIL import Image
>>> Image.core.jpeglib_version
'9.0'
>>> Image.core.jp2klib_version
'2.3.1'
```

**Comment by radarhere** [comment]
This is sounding like it's a problem specific to your machine, as I don't recall any other issues discussing not being able to read any JPEGs at all. Certainly this isn't a universal problem.

So if you're interested in continuing this discussion, then the issue can be re-opened, but otherwise, I don't think there's anything to do here.

**Comment by duhaime** [comment]
Why not leave this open to determine what's wrong with OP's machine with respect to PIL?

**Comment by hugovk** [comment]
Because the OP hadn't responded to the April 2020 question in a month, and also didn't express interest in keeping it open (May 2020 closure comment).

As we have hundreds of open issues, if there's no clear problem and we can't reproduce it, plus the OP doesn't reply, issues will be closed to try and keep the issue tracker manageable.

If you have a similar problem, please could you open a new issue with details? Thank you!

**Comment by duhaime** [comment]
Amen, that's understandable!
